src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/DataContent.cs

```diff
@@ -256,4 +259,120 @@ private string DebuggerDisplay
                 $"Data = {uri.Substring(0, MaxLength)}...";
         }
     }
+
+    /// <summary>The default media type for unknown file extensions.</summary>
+    private const string DefaultMediaType = "application/octet-stream";
+
+    /// <summary>
+    /// Loads a <see cref="DataContent"/> from a file path asynchronously.
+    /// </summary>
+    /// <param name="path">The file path to load the data from.</param>
+    /// <param name="mediaType">
+    /// The media type (also known as MIME type) represented by the content. If not provided,
+    /// it will be inferred from the file extension. If it cannot be inferred, "application/octet-stream" is used.
+    /// </param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests.</param>
+    /// <returns>A <see cref="DataContent"/> containing the file data with the inferred or specified media type and name.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="path"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="path"/> is empty.</exception>
+    public static async ValueTask<DataContent> LoadFromAsync(string path, string? mediaType = null, CancellationToken cancellationToken = default)
+    {
+        _ = Throw.IfNullOrEmpty(path);
+
+        // Infer media type from the file extension if not provided
+        mediaType ??= System.Net.Mime.MediaTypeMap.GetMediaType(path) ?? DefaultMediaType;
+
+        // Read the file contents
+        byte[] data = await File.ReadAllBytesAsync(path, cancellationToken).ConfigureAwait(false);
+
+        string? name = Path.GetFileName(path);
+        return new DataContent(data, mediaType)
+        {
+            Name = string.IsNullOrEmpty(name) ? null : name
+        };
```

Comment:
Wonder if we could just have this as an extension that uses the `Stream`.

```suggestion
        _ = Throw.IfNullOrEmpty(path);

        using var fileStream = File.OpenRead(path);
        return await LoadFromAsync(fileStream, mediaType, cancellationToken).ConfigureAwait(false);
```